packages/codecs-data-structures/README.md

```diff
@@ -16,10 +16,488 @@
 
 This package contains codecs for various data structures such as arrays, maps, structs, tuples, enums, etc. It can be used standalone, but it is also exported as part of the Solana JavaScript SDK [`@solana/web3.js@experimental`](https://github.com/solana-labs/solana-web3.js/tree/master/packages/library).
 
-## Types
+This package is also part of the [`@solana/codecs` package](https://github.com/solana-labs/solana-web3.js/tree/master/packages/codecs) which acts as an entry point for all codec packages as well as for their documentation.
 
-TODO
+## Array codec
 
-## Functions
+The `getArrayCodec` function accepts any codec of type `T` and returns a codec of type `Array<T>`. For instance, here’s how we can create a codec for arrays of numbers that each fit in a single byte.
 
-TODO
+```ts
+const bytes = getArrayCodec(getU8Codec()).encode([1, 2, 3]);
+const array = getArrayCodec(getU8Codec()).decode(bytes);
+```
+
+By default, the size of the array is stored as a `u32` prefix before encoding the items.
+
+```ts
+getArrayCodec(getU8Codec()).encode([1, 2, 3]);
+// 0x03000000010203
+//   |       └-- 3 items of 1 byte each.
+//   └-- 4-byte prefix telling us to read 3 items.
+```
+
+However, you may use the `size` option to configure this behaviour. It can be one of the following three strategies:
+
+-   `Codec<number>`: When a number codec is provided, that codec will be used to encode and decode the size prefix — much like the `u32` example above.
```

Comment:
Nit: it's a bit confusing what "the `u32` example above" refers to, because the u32 isn't in the code example. Would it be clear to say `U32Codec` provides the default `u32` prefix described above?

Reply:
You're right, that was a bit confusing to read. I removed that part of the sentence altogether because we're already mentioning the `u32` default prefix before reaching this bullet point.